libs/portals/admin/ids-admin/src/components/forms/AddPermissions/AddPermissions.tsx

```diff
@@ -8,125 +8,167 @@ import {
 } from '@island.is/island-ui/core'
 import { m } from '../../../lib/messages'
 import { useLocale } from '@island.is/localization'
-import React from 'react'
-import { Form } from 'react-router-dom'
+import React, { useContext } from 'react'
 import { ShadowBox } from '../../ShadowBox/ShadowBox'
+import { ClientContext } from '../../../shared/context/ClientContext'
+import { AuthAdminClientAllowedScope } from '@island.is/api/schema'
+import { getTranslatedValue } from '@island.is/portals/core'
+import { useGetAvailableScopesQuery } from './AvailableScopes.generated'
+import { useParams } from 'react-router-dom'
 
 interface AddPermissionsProps {
   isVisible: boolean
-
-  onClose(): void
+  onClose: () => void
+  onAdd: (permissions: AuthAdminClientAllowedScope[]) => void
 }
 
-type Permission = {
-  id: string
-  label: string
-  description: string
-  api: string
-  locked?: boolean
-}
+function AddPermissions({ isVisible, onClose, onAdd }: AddPermissionsProps) {
+  const { formatMessage, locale } = useLocale()
+  const [selected, setSelected] = React.useState<
+    Map<string, AuthAdminClientAllowedScope>
+  >(new Map())
+  const params = useParams()
+  const [availableScopes, setAvailableScopes] = React.useState<
+    AuthAdminClientAllowedScope[]
+  >([])
+
+  const {
+    selectedEnvironment,
+    setAddedScopes,
+    addedScopes,
+    removedScopes,
+    setRemovedScopes,
+  } = useContext(ClientContext)
+
+  const { loading } = useGetAvailableScopesQuery({
```

Comment:
You can get the data from the query and use then component so you don't need the local availableScope state
```suggestion
  const {data, loading } = useGetAvailableScopesQuery({
```